src/java.desktop/share/classes/sun/java2d/marlin/TransformingPathConsumer2D.java

```diff
@@ -898,22 +899,22 @@ public long getNativeConsumer() {
         final float[] clipRect;
 
         // clip rectangle (ymin, ymax, xmin, xmax) including padding:
-        final float[] clipRectPad = new float[4];
+        final double[] clipRectPad = new double[4];
```

Comment:
It does not seem related to this fix? Or an updated test uncovered this?

Reply:
This fix in Marlin (float) renderer is needed as this numerical precision issue was triggered by the StrokedLinePerf using the MarlinRenderingEngine. It is very similar to DTransformingPathConsumer2D (but reuse was not trivial).

I propose to remove the MarlinRenderingEngine (single-precision) in a follow-up issue ... to simplify the maintenance.